aten/src/ATen/native/mps/operations/UnaryKernel.mm

```diff
@@ -0,0 +1,135 @@
+#define TORCH_ASSERT_ONLY_METHOD_OPERATORS
+#include <ATen/mps/MPSProfiler.h>
+#include <ATen/native/DispatchStub.h>
+#include <ATen/native/UnaryOps.h>
+#include <ATen/native/mps/OperationUtils.h>
+#include <ATen/native/mps/UnaryConstants.h>
+#include <torch/mps.h>
```

Comment:
I think this header is no longer needed?

Reply:
> I think this header is no longer needed?

I need <torch/mps.h> to call [torch::mps::synchronize()](https://github.com/pytorch/pytorch/blob/c72e5ca2b79583336271d38ee754e5dc6a62e087/aten/src/ATen/native/mps/operations/UnaryKernel.mm#L131) to make sure the kernel is done with its work before I return. For large tensor size, e.g. 100k elements,  I found that it returns too early before the operation is completed and would not give me the right result. Would love to get feedback if there's a way for me to avoid doing so.

Reply:
I'm curious how you checked that the results were incorrect with a large tensor size. Can you please provide a minimal repro?

Reply:
copy this gist into a python notebook celll and play it: https://gist.github.com/TaiPhamD/d781fe01fe2a6a37c71fa8b2058bac98

## without synchronization
without the synchronization the cell will never complete:
```
CPU torch.erfinv time:  1.3201049000000011
MPS torch.erfinv time:  1.30214046659994e-06
MPS torch.erfinv is 101379607.95020592 percent faster than CPU torch.erfinv
```

The mps torch.erfinv get some wild fast run time meaning it returned too early.

## with synchronization
This is the output with synchronization:


```
CPU torch.erfinv time: 2.654937833400254
MPS torch.erfinv time: 0.009831255332002912
MPS torch.erfinv is 27005.07456822776 percent faster than CPU torch.erfinv
length of y_mps: 199999992, length of y_cpu: 199999992, length of x: 199999992
MSE between MPS and CPU torch.erfinv: tensor(4.2339e-14)
Largest difference
x: -0.9999980330467224, y_cpu: -3.363569736480713, y_mps: -3.3635685443878174 , diff = -1.1920928955078125e-06
```

Reply:
Thank you, will look into it later.

Reply:
I checked out and compiled your PR with `torch::mps::synchronize();` commented out. It works as expected without problem. I don't have notebook installed so I removed the notebook specific codes:

```python
def test_erfinv():
    import torch
    x = torch.arange(-1, 1, 1e-8) # default cpu tensor
    #measure CPU compute time by calling torch.erfinv
    #time = #%timeit -o -q -r 5 torch.erfinv(x)
    cpu_time = 123#time.average
    print("CPU torch.erfinv time: ", cpu_time)
    x = x.to("mps")
    # measure MPS compute time
    #time = %timeit -o -q -r 5 torch.erfinv(x)
    mps_time = 123#time.average
    print("MPS torch.erfinv time: ", mps_time)
    print(f"MPS torch.erfinv is {cpu_time/mps_time*100} percent faster than CPU torch.erfinv")

    # compute MSE between MPS and CPU torch.erfinv
    x = x.to("cpu")
    y_cpu = torch.erfinv(x)
    x = x.to("mps")
    y_mps = torch.erfinv(x)
    y_mps = y_mps.to("cpu")
    mask = torch.isfinite(y_cpu) & torch.isfinite(y_mps.to("cpu"))
    y_mps = y_mps[mask]
    y_cpu = y_cpu[mask]
    x = x[mask]
    print(f"length of y_mps: {len(y_mps)}, length of y_cpu: {len(y_cpu)}, length of x: {len(x)}")
    mse = torch.square(y_cpu - y_mps).mean()
    print("MSE between MPS and CPU torch.erfinv: ", mse)
    diff = torch.abs(y_cpu - y_mps)
    print("Largest difference")
    print(f"x:  {x[torch.argmax(diff)]}, y_cpu: {y_cpu[torch.argmax(diff)]}, y_mps: {y_mps[torch.argmax(diff)]} , diff = {y_cpu[torch.argmax(diff)] - y_mps[torch.argmax(diff)]}")
```

```
CPU torch.erfinv time:  123
MPS torch.erfinv time:  123
MPS torch.erfinv is 100.0 percent faster than CPU torch.erfinv
length of y_mps: 199999992, length of y_cpu: 199999992, length of x: 199999992
MSE between MPS and CPU torch.erfinv:  tensor(4.2339e-14)
Largest difference
x:  -0.9999980330467224, y_cpu: -3.363569736480713, y_mps: -3.3635685443878174 , diff = -1.1920928955078125e-06
```
Can you please try running the code in a pure Python runtime (not sure if the problem is related to notebook)?
Also, `DispatchStub.h` looks unnecessary as well.

Reply:
@qqaatw  yes you're correct it is related the magic command %timeit. I tried manually looping using a for loop instead of using the %timeit and it works. Also without making the cpu blocking call synchronization the speed up is also significantly faster so we definitely don't want to use synchronization unless we absolutely have to. 

## without synchronization but avoiding using %timeit and use for loop instead:
```
CPU torch.erfinv time:  1.3240717887878417
MPS torch.erfinv time:  0.0004521846771240234
MPS torch.erfinv is 292816.5981229569 percent faster than CPU torch.erfinv
length of y_mps: 199999992, length of y_cpu: 199999992, length of x: 199999992
MSE between MPS and CPU torch.erfinv:  tensor(4.1816e-14)
Largest difference
x:  -0.9999992847442627, y_cpu: -3.5052666664123535, y_mps: -3.505267858505249 , diff = 1.1920928955078125e-06
```

@kulinseth  Do you think the issue with %timeit is out of scope for this change?

Reply:
Since the MPS tests are green, I would be inclined to merging this PR first to unlock users.

We can investigate it further and submit a follow-up PR if we find a fix for that.